src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -112,12 +126,27 @@ mongoc_client_pool_new (const mongoc_uri_t *uri)
 
 #ifndef MONGOC_ENABLE_SSL
    if (mongoc_uri_get_tls (uri)) {
-      MONGOC_ERROR ("Can't create SSL client pool,"
-                    " SSL not enabled in this build.");
+      bson_set_error (error,
+                      MONGOC_ERROR_COMMAND,
+                      MONGOC_ERROR_COMMAND_INVALID_ARG,
+                      "Can't create SSL client pool, SSL not enabled in this "
+                      "build.");
```

Comment:
I missed this in a previous review, but there is no reason for `mongoc_client_pool_new_with_error` to log an error directly. `mongoc_client_pool_new` will already do so (preserving behavior) so this should merely set the error on the `bson_error_t` to allow the caller to interpret it as they wish.